--- coreutils-9.7/src/yes.c	2026-01-01 00:00:00.000000000 +0000
+++ coreutils-9.7/src/yes.c	2026-08-16 00:00:00.000000000 +0000
@@ -89,10 +89,11 @@
     }
   while (++operandp < operand_lim);
 
-  /* Improve performance by using a buffer size greater than BUFSIZ / 2.  */
-  if (bufalloc <= BUFSIZ / 2)
+  /* Improve performance by using a buffer size greater than half of
+     the sequential write plateau (256 KiB).  */
+  if (bufalloc <= (256 * 1024) / 2)
     {
-      bufalloc = BUFSIZ;
+      bufalloc = 256 * 1024;
       reuse_operand_strings = false;
     }
 
--- coreutils-9.7/src/tr.c	2026-01-01 00:00:00.000000000 +0000
+++ coreutils-9.7/src/tr.c	2026-08-16 00:00:00.000000000 +0000
@@ -241,7 +241,7 @@
    It is set in main and used there and in validate().  */
 static bool translating;
 
-static char io_buf[BUFSIZ];
+static char io_buf[256 * 1024];
 
 static char const *const char_class_name[] =
 {
--- coreutils-9.7/src/tee.c	2026-01-01 00:00:00.000000000 +0000
+++ coreutils-9.7/src/tee.c	2026-08-16 00:00:00.000000000 +0000
@@ -228,7 +228,7 @@
   size_t n_outputs = 0;
   FILE **descriptors;
   bool *out_pollable IF_LINT ( = nullptr);
-  char buffer[BUFSIZ];
+  static char buffer[256 * 1024];
   ssize_t bytes_read = 0;
   int i;
   int first_out = 0;  /* idx of first non-null output in descriptors */
--- coreutils-9.7/src/tac.c	2026-01-01 00:00:00.000000000 +0000
+++ coreutils-9.7/src/tac.c	2026-08-16 00:00:00.000000000 +0000
@@ -58,10 +58,10 @@
 
 
 /* The number of bytes per atomic read. */
-#define INITIAL_READSIZE 8192
+#define INITIAL_READSIZE (256 * 1024)
 
 /* The number of bytes per atomic write. */
-#define WRITESIZE 8192
+#define WRITESIZE (256 * 1024)
 
 /* The string that separates the records of the file. */
 static char const *separator;
